tests/js/searchtools.spec.js

```diff
@@ -95,7 +95,21 @@ describe('Basic html theme search', function() {
       ]];
       expect(Search.performTermsSearch(searchterms, excluded, terms, titleterms)).toEqual(hits);
     });
+  });
 
+  describe('unicode normalization', function() {
+    it('should parse queries with half-width and full-width characters equivalently', function() {
+      const orig = DOCUMENTATION_OPTIONS.SEARCH_UNICODE_NORMALIZATION;
+      try {
+        DOCUMENTATION_OPTIONS.SEARCH_UNICODE_NORMALIZATION = 'NFKC';
+        const halfWidthQuery = Search._parseQuery('Sphinx');
+        const fullWidthQuery = Search._parseQuery('Ｓｐｈｉｎｘ');
+
+        expect(halfWidthQuery).toEqual(fullWidthQuery);
+      } finally {
+        DOCUMENTATION_OPTIONS.SEARCH_UNICODE_NORMALIZATION = orig; // restore
```

Comment:
Hmm.  Copying and then restoring the normalization option here seems inelegant.  This may be a limitation of the test suite at the moment -- the fixtures vary for each test, but the documentation options do not.

In addition, I am considering how this might interact with #13395 or similar features in future.

I think the arrangement is OK for the moment -- but perhaps we should consider extracting a separate `Search._normalizeQuery` function before `_parseQuery` occurs.

What do you think @tokuhirom?

Reply:
at first, the root cause is the Search module depends on the global variable. It makes hard to do the unit testing.
I mean, if sphinx have an API like this, 

```
<script src="searchtools.js"></script>
<script>
new Search({
  fileSuffix: '{{ file_suffix }}',
  ...
}).initialize();
</script>
```

I suggest to rewrite the APIs like this. but so, it's the too big deal.

at the moment, i moved the cleanup function to `afterEach` block. and split the `_normalizeQuery` method.

Reply:
That sounds good to me; thank you.

With those changes, the `def normalize` function in the Python code will be a mirror of the `function _normalizeQuery` in the `searchtools.js` code.  So therefore: it might be worth adding a comment in the Python code to explain that (but not in the JS code -- because not everyone reading the JS code will be able to view the Python code).

Reply:
PS: I think an additional `git push` may be required (your explanation makes sense, but I don't find the updated commits in the branch at the moment)

Reply:
oops. pushed.